src/main/java/com/diffblue/deeptestutils/CompareWithFieldList.java

```diff
@@ -0,0 +1,226 @@
+package com.diffblue.deeptestutils;
+
+import java.io.FileNotFoundException;
+import java.io.PrintStream;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.HashSet;
+
+/**
+ * <code>CompareWithFieldList</code> does a deep compare on objects and their
+ * members.
+ *
+ * @author <a href="http://diffblue.com">Diffblue</a>
+ */
+public final class CompareWithFieldList {
+
+  /**
+   * Private constructor to prevent instantiation of the class with static
+   * methods.
+   */
+  private CompareWithFieldList() {
+  }
+
+  /**
+   * Output stream for debugging.
+   */
+  private static PrintStream debugOut;
+
+  static {
+    String outPath = System.getenv("CWFL_OUTPUT_FILE");
+    if (outPath != null) {
+      try {
+        debugOut = new PrintStream(outPath);
+      } catch (FileNotFoundException e) {
+        throw new RuntimeException(e);
+      }
+    } else {
+      debugOut = null;
```

Comment:
Suggested tidy up, not important to do right now - We could do the following instead of setting `debugOut = null`, in order to avoid checking it for null everytime later on -
```
	debugOut = new java.io.PrintStream(
	    new java.io.OutputStream() {
	       public void write(int b){}
	    } );
```